### ISSUE TITLE
Can't repeat (with .) action

### ISSUE THREAD

**Issue by houshuang** [timcharper/evil-surround#24]
This package is absolutely brilliant, and makes me feel like a code editing ninja. However, it doesn't seem to add the action to Evil's list of past commands - for example, I might have a list of numbers that need to be enclosed in parenthesis, and try to do ysiw" on one of them, and do the rest with move and . (inspired by Practical Vim). In this case, . repeats the action I did before ysiw", not ysiw". I've tried different surround commands, and this is always the case.

**Comment by exitface** [comment]
+1

**Comment by Dickby** [comment]
+1

**Comment by sooheon** [comment]
Coming from vim recently this is the thing I missed immediately.

**Comment by duerrp** [comment]
+1evil

P.S. thanks for the great work so far!

**Comment by igagen** [comment]
+1 this would be a great addition!

**Comment by ryanorendorff** [comment]
+1. Anything I could do to assist? I'm an emacs newbie at the moment.

**Comment by unhammer** [comment]
https://bytebucket.org/lyro/evil/raw/default/doc/evil.pdf says that evil-define-operator has `:repeat t` by default. Anyway, I tried adding `:repeat t` to evil-define-operator evil-surround-region, but nothing happens. I also tried both `(evil-declare-repeat #'evil-surround-region)`  and `(evil-declare-change-repeat #'evil-surround-region)`, but no luck, it still just repeats the preceding action (e.g. C-M-x since I just eval'd the thing).

I fear the functions themselves will have to surround their changes with `(evil-repeat-start-record-changes) … (evil-repeat-record-change relpos ins ndel)  … (evil-repeat-finish-record-changes)`.
